src/stream_engine.cpp

```diff
@@ -908,9 +910,25 @@ void zmq::stream_engine_t::mechanism_ready ()
             return;
         }
         errno_assert (rc == 0);
-        _session->flush ();
+        flush_session = true;
     }
 
+    if (_options.router_notify & ZMQ_NOTIFY_CONNECT) {
+        msg_t connector;
```

Comment:
Maybe rename to connect_notification?

Reply:
I now saw that this is copied from the ZMQ_STREAM_NOTIFY handling, so better leave it as it is.